### PR TITLE
Support a toggle on Nomad client mTLS

### PR DESCRIPTION
**Pull request by ryanwohara**
SERVER-613

This PR creates a way for a user to disable mTLS. This is a high priority task for the RE team.

Manually changing `enabled` under the `nomad_tls` module in either EKS or GKE will disable mTLS. There is a warning that it is risky and not recommended.

### CODE REVIEW

File: eks/nomad/main.tf [hunk 1]
```diff
@@ -15,11 +15,12 @@ resource "aws_key_pair" "ssh_key" {
 module "nomad_tls" {
   source   = "../../shared/modules/tls"
   basename = var.basename
+  enabled  = 1 # It can be dangerous to modify this value
```
Comment (robinsturmeit) [inline eks/nomad/main.tf:18]: I think you can save a lot of work using [count](https://www.terraform.io/docs/configuration/meta-arguments/count.html) here and maybe use a variable that can be conveniently set via `terraform.tfvars` or cli on the top level. Then you can leave the TLS module as-is.

-- same applies to GKE

**Review by robinsturmeit — CHANGES_REQUESTED** [review]
Had a suggestion for an improvement. Furthermore, are you planning a separate PR for SHC for the server-side config?

File: eks/nomad/variables.tf [hunk 2]
```diff
@@ -41,3 +41,9 @@ variable "ssh_key" {
   default     = null
   description = "SSH key to authenticate access to Nomad clients. If not set, SSH is disabled"
 }
+
+variable "enable_mtls" {
+  type        = number
```
Comment (robinsturmeit) [inline eks/nomad/variables.tf:46]: nit: why is this a number and not a bool?

**Review by robinsturmeit — COMMENTED** [review]
Okay, looking good. Two questions before I approve:
1. Why are the variables numbers and not bools?
2. Are you positive that having an empty tls stanza is causing no issues with the nomad clients?

**Comment by ryanwohara** [comment]
1. Updated numbers to bools
2. What empty tls stanza? You may have misread - I omit the stanza entirely.

**Comment by robinsturmeit** [comment]
> 1. Updated numbers to bools
> 2. What empty tls stanza? You may have misread - I omit the stanza entirely.

Ah, yes, you're right. Approved, thank you for going along with my requests :)

**Review by robinsturmeit — APPROVED** [review]

**Review by smcelmurry — APPROVED** [review]
I love you

**Pull request merged**